src/components/fields/DataSelector.js

```diff
@@ -45,7 +45,8 @@ export class UnconnectedDataSelector extends Component {
             'carpet',
             'contourcarpet',
           ].includes(props.container.type)) ||
-        (props.container.type === 'table' && props.attr !== 'columnorder');
+        (props.container.type === 'table' && props.attr !== 'columnorder') ||
+        Array.isArray(this.fullValue);
```

Comment:
well, I don't think all this.fullValues that are arrays are 2d..
2d means an array of arrays, 
so, I would check that if fullValue is an array, that all the elements in it are arrays too..

Reply:
No, this.fullValue is made of datasources, not data.
So for tables, contour Z, etc. it looks like ['ints', 'big ints'...]
Would it be different in cloud?

Reply:
oh ok, so you're saying that when only one column can be selected, this.fullValue will not be an array? it will just be a string?

if that's the case, then ok, yes, I approuve of this logic change

Reply:
yep, it's a string like "ints"

Reply:
ok 💃 thanks!